### PR TITLE
refactor(router) code clean with utils.lua

### PR DESCRIPTION
**Pull request by chronolaw**
### Summary

Remove redundant code in `atc_compat.lua` and `traditional.lua`.

### CODE REVIEW

File: kong/router/utils.lua [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+    end
+
+    if service.name then
+      header["Kong-Service-Name"] = service.name
+    end
+  end
+end
+
+
+local function get_upstream_uri(matched_route, request_postfix, req_uri,
```
Comment (dndx) [inline kong/router/utils.lua:131]:
```suggestion
local function get_upstream_uri_v0(matched_route, request_postfix, req_uri,
```

or something similar to make it obvious that this only supports `v0` of `path_handling`.

File: kong/router/utils.lua [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+  if sni and type(sni) ~= "string" then
+    error("sni must be a string", 2)
+  end
+  if req_headers and type(req_headers) ~= "table" then
+    error("headers must be a table", 2)
+  end
+end
+
+
+local function debug_http_headers(var, header, match_t)
```
Comment (dndx) [inline kong/router/utils.lua:102]:
```suggestion
local function add_debug_headers(var, header, match_t)
```

is probably better, `debug_http_headers` sounds a bit strange,

File: kong/router/utils.lua (hunk 1 above)
Comment (chronolaw) [reply kong/router/utils.lua]: done

File: kong/router/utils.lua (hunk 2 above)
Comment (chronolaw) [reply kong/router/utils.lua]: done

**Review by dndx — APPROVED** [review]

**Pull request merged**